packages/xo-web/src/common/xo/index.js

```diff
@@ -1226,41 +1232,42 @@ export const startVms = vms =>
     }
   }, noop)
 
-export const stopVm = async (vm, force = false) => {
-  try {
-    await confirm({
-      title: _('stopVmModalTitle'),
-      body: _('stopVmModalMessage', { name: vm.name_label }),
-    })
-
-    return await _call('vm.stop', { id: resolveId(vm), force })
-  } catch (error) {
-    if (error === undefined) {
-      return
-    }
+export const stopVm = async (vm, hardShutdown = false) => {
+  let forceBlockedOperation = false
+  const id = resolveId(vm)
 
-    if (!vmLacksFeature.is(error) || force) {
-      throw error
-    }
+  await confirm({
+    title: _('stopVmModalTitle'),
+    body: _('stopVmModalMessage', { name: vm.name_label }),
+  })
 
-    try {
-      await confirm({
-        title: _('vmHasNoTools'),
-        body: (
-          <div>
-            <p>{_('vmHasNoToolsMessage')}</p>
-            <p>
-              <strong>{_('confirmForceShutdown')}</strong>
-            </p>
-          </div>
-        ),
-      })
-    } catch {
-      return
-    }
+  return retry(() => _call('vm.stop', { id, force: hardShutdown, forceBlockedOperation }), {
```

Comment:
If the error is undefined, we don't need to retry, right? If so, you need to add this case.

Reply:
why the error could be undefined?